modules/dynamic-plugins/proc-export-third-party-plugins-rhdh.adoc

```diff
@@ -0,0 +1,124 @@
+
+* *Embedded dependencies* are bundled into the dynamic plugin package with their dependencies hoisted to the top level. By default, packages with `-node` or `-common` suffixes are embedded.
++
+You can use the `--embed-package` flag to specify additional embedded packages. For example, packages from the same workspace that do not follow the default naming convention.
++
+The following is an example of exporting a dynamic plugin with shared and embedded packages:
++
+.Example dynamic plugin export with shared and embedded packages
+[source,bash]
+----
+npx @janus-idp/cli@latest export-dynamic-plugin --shared-package '!/@backstage/plugin-notifications/' <1> --embed-package @backstage/plugin-notifications-backend <2>
+----
++
+<1> `@backstage/plugin-notifications` package is treated as a private dependency and is bundled in the dynamic plugin package, despite being in the `@backstage` scope.
+<2> `@backstage/plugin-notifications-backend` package is marked as an embedded dependency and is bundled in the dynamic plugin package.
+
+Front-end plugins::
++
+Front-end plugins can use `scalprum` for configuration, which the CLI can generate automatically during the export process. The generated default configuration is logged when running the following command:
++
+.Example command to log the default configuration
+[source,bash]
+----
+npx @janus-idp/cli@latest export-dynamic
+----
++
+The following is an example of default `scalprum` configuration:
++
+.Default `scalprum` configuration
+[source,json]
+----
+"scalprum": {
+  "name": "<package_name>",  // The Webpack container name matches the NPM package name, with "@" replaced by "." and "/" removed.
+  "exposedModules": {
+    "PluginRoot": "./src/index.ts"  // The default module name is "PluginRoot" and doesn't need explicit specification in the app-config.yaml file.
+  }
+}
+----
++
+You can add a `scalprum` section to the `package.json` file. For example:
++
+.Example `scalprum` customization
+[source,json]
+----
+"scalprum": {
+  "name": "custom-package-name",
+  "exposedModules": {
+    "FooModuleName": "./src/foo.ts",
+    "BarModuleName": "./src/bar.ts"
+    // Define multiple modules here, with each exposed as a separate entry point in the Webpack container.
+  }
+}
+----
++
+Dynamic plugins might need adjustments for {product-short} needs, such as static JSX for mountpoints or dynamic routes. These changes are optional but might be incompatible with static plugins.
++
+To include static JSX, define an additional export and use it as the dynamic plugin's `importName`. For example:
++
+.Example static and dynamic plugin export
+[source,tsx]
+----
+// For a static plugin
+export const EntityTechdocsContent = () => {...}
+
+// For a dynamic plugin
+export const DynamicEntityTechdocsContent = {
+  element: EntityTechdocsContent,
+  staticJSXContent: (
+    <TechDocsAddons>
+      <ReportIssue />
+    </TechDocsAddons>
+  ),
+};
+----
+--
+
+.Procedure
```

Comment:
From this line to the end of this file, should be right after Prerequisites and before backend plugins

Reply:
@kadel According to the procedural module template, the module should conclude with a procedure. The details about backend and frontend plugins include some conceptual information, so I think adding them to the prerequisites section would help clarify these requirements for users. Therefore, we’ll retain them in the prerequisites.